network-area-diagram-viewer/src/main/resources/svg.js

```diff
@@ -0,0 +1,463 @@
+
+    function classIsDraggable(element) {
+        // For svg objects it is better to get class names through attribute, to avoid potential SVGAnimatedString values
+        // https://developer.mozilla.org/en-US/docs/Web/API/Element/className
+        var className = element.getAttribute("class");
+        if (!className) {
+            var parent = element.parentElement;
+            if (parent) {
+                className = parent.getAttribute("class");
+            }
+        }
+        return className && (className.includes("vl-node") || className.includes("boundary-node") || className.includes("busnode"));
+    }
+
+    function hasId(element){
+        return typeof element.id != 'undefined' && element.id != '';
+    }
+
+    function update(svgElem, position, translation) {
+        var updateOnServer = false;
+        if (updateOnServer) {
+            updateOnServer();
+        } else {
+            // Adjust the id of the element that has been moved
+            // It we were moving a bus node (its circle),
+            // find the id of the corresponding voltage level
+            var id = svgElem.getAttribute("id");
+            var parent = svgElem.parentElement;
+            var idParent = parent.getAttribute("id");
+            if (idParent) {
+                id = idParent;
+            }
+            updateOnClient(id, position, translation);
+        }
+    }
+
+    function updateOnClient(id, position, translation) {
+        console.log("updateOnClient, moved node id = " + id);
+        translateText(id, translation);
+        // TODO(Luma) Update here the position of all inner busNodes
+        for (var edge of svg.getElementsByTagName("nad:edge")) {
+            var node1 = edge.getAttribute("node1");
+            var node2 = edge.getAttribute("node2");
+            console.log("  consider edge (" + node1 + ", " + node2 + ") after moved node " + id);
+            if (node1 === id || node2 === id) {
+                updateEdge(edge, node1, node2, id, position, translation);
+            }
+        }
+    }
+
+    function updateEdge(edgeMetadata, node1, node2, movedNodeId, position, translation) {
+        const DEBUG_EXPERIMENTS = false;
+        if (DEBUG_EXPERIMENTS) {
+            updateEdgeExperiments(edgeMetadata, node1, node2, movedNodeId, position, translation);
+        } else {
+            updateEdgeUsingRotation(edgeMetadata, node1, node2, movedNodeId, position, translation);
+        }
+    }
+
+    function updateEdgeUsingRotation(edgeMetadata, node1, node2, movedNodeId, position, translation) {
+        // This edge is adjacent to the moved node
+        console.log("  adjacent edge (" + node1 + ", " + node2 + ")");
+        var edgeId = edgeMetadata.getAttribute("svgid");
+        var edgeSvg = svg.getElementById(edgeId);
+        if (!edgeSvg) {
+            return;
+        }
+
+        // Translate all edge-related graphical elements to the new position
+        translate(svg, edgeSvg, translation);
```

Comment:
I'd do that translate only in an else below, for the null case `otherNodeSvg`, and compute the whole transform matrix otherwise in `rotateEdge`

Reply:
Agree, another to do.

Reply:
followed your suggestion, done.